rust/hyperlane-core/src/db/hyperlane_db.rs

```diff
@@ -198,50 +200,89 @@ impl HyperlaneDB {
         self.store_gas_payment_meta_processed(meta)?;
 
         // Update the total gas payment for the message to include the payment
-        self.update_gas_payment_for_message_id(&gas_payment_with_meta.payment)?;
+        self.update_gas_payment_for_message_id(gas_payment_with_meta.payment)?;
 
         // Return true to indicate the gas payment was processed for the first time
         Ok(true)
     }
 
-    /// Record a gas payment, identified by its metadata, as processed
-    fn store_gas_payment_meta_processed(
+    /// Processes the gas expenditure and store the total expenditure for the message.
+    pub fn process_gas_expenditure(
         &self,
-        gas_payment_meta: &InterchainGasPaymentMeta,
+        gas_expenditure_with_meta: &GasExpenditureWithMeta,
```

Comment:
I see that the meta part isn't used, any reason we pass that in here?

Reply:
Missed that, was used at a version some commit ago. Removed.